tests/test_watcher_comprehensive.py

```
@@ -0,0 +1,574 @@
+        """Test successful initialization."""
+        watcher = GengoWatcher(mock_config, mock_state, logger)
+        assert watcher.config == mock_config
+        assert watcher.state == mock_state
+        assert watcher.logger == logger
+        assert watcher.start_time > 0
+        assert watcher.session_new_entries == 0
+        assert watcher.websocket_status == "Disabled"
+
+    def test_initialization_validates_check_interval(self, mock_config, mock_state, logger):
+        """Test that check_interval is validated."""
+        mock_config.get.side_effect = lambda s, k, **kw: 0 if (s, k) == ("Watcher", "check_interval") else mock_config.config.get(s, {}).get(k)
```

Ruff (0.14.14)

Unused lambda argument: `kw`

(ARG005)